networklibrary/src/main/java/com/eis/communication/network/NetDictionary.java

```diff
@@ -0,0 +1,32 @@
+package com.eis.communication.network;
+
+/**
+ * Interface to define a network dictionary
+ * @param <K> The Key for each resource
+ * @param <R> The Resource
+ *
+ * @author Marco Cognolato
+ */
+public interface NetDictionary<K,R>{
+
+    /**
+     * Adds a resource to the network dictionary
+     * @param key The key which defines the resource
+     * @param resource The resource to add
+     */
+    void addResource(K key, R resource);
+
+    /**
+     * Removes a resource from the dictionary
+     * @param key The key which defines the resource
+     */
```

Comment:
I think you should specify "to be removed" on @param specification, as you wrote in the getResource method.
Same also in addResource method.